### ISSUE TITLE
Accecibility notes on navs

### ISSUE THREAD

**Issue by andresgalante** [twbs/bootstrap#23831]
Navs have a section for [accessibility notes](https://getbootstrap.com/docs/4.0/components/navs/#regarding-accessibility).

Since most users will just copy and paste the code snippets and hardly anyone will actually read the docs I wonder if we should bake accessibility into every code snippet.

I know it will be way more verbose and it might not show the point of the example, but the dent Bootstrap can make just be changing this would be huge!

@patrickhlauke What do you think? If you like this idea I can work on a PR

**Comment by Fouzyyyy** [comment]
The examples provided in the docs mostly aim to provide twbs users the visual output of a given class/component; hence accessibility is out of the scope for this kind of usage; isn't it?

**Comment by andresgalante** [comment]
@Fouzyyyy you are right, the examples on navs are to provide class information. But I couldn't disagree more with it. We have accessibility roles and aria tags all over the place on other parts of the docs and @patrickhlauke is doing an amazing job trying to make things accessible.

Accessibility should be one of the main concerns of any front end implementation, if we can get baked in accessible snippets we will be sure that the "copy and paste" users of the docs will end up with an accessible website.

That is a HUGE impact due to the amount of bootstrap users.

An accessible web is a better web, and we can make a difference just by making snippets accessible.

**Comment by patrickhlauke** [comment]
Feel free to submit a PR (which I guess for the `<ul>` based examples will simply mean having a `<nav>` or `<div role="navigation">` wrapper...)

**Comment by Fouzyyyy** [comment]
@andresgalante I understand, ideally the accessibility must be taken into account in any piece of code, whether it's an example code or production one.

Are you submitting a PR? did you identify all the docs examples that must be altered? I can help with this if needed.

**Comment by andresgalante** [comment]
Hi @Fouzyyyy thanks a lot for the offer 😄  I did submit a PR yesterday but it was wrongly linked to the issue, I just fixed the reference.

**Comment by patrickhlauke** [comment]
In general, always happy to see PRs that improve things where they may be lacking :)

I would suggest splitting any effort up into small chunks, though (rather than working on a massive single pull request). Makes it easier to review/discuss on a case by case basis if there's any doubt, or if it comes down to opinion/taste rather than being a clear error/omission. And means that smaller good additions can be merged more quickly without being held up by those other ones.

**Comment by mdo** [comment]
Should this be closed for more specific issues?

**Comment by patrickhlauke** [comment]
yes i think it would be best to have case-specific issues/PRs here - makes it a bit clearer.

**Comment by patrickhlauke** [comment]
(and yes, if the main concern was that primary navigations should be given a `role="navigation"` or be wrapped with a `<nav>`, it's perhaps dangerous to include that in the various examples, as then you may end up with the opposite effect - people adding those everywhere, even when what they're using them for is NOT primary navigation. which is not a major problem per se, but if a page has more than one navigation landmark, it immediately reduces the usefulness of such a landmark to assistive tech users, as it makes it MORE difficult for them to work out which one exactly is the primary navigation)
